react/the_react_ecosystem/react_router.md

```diff
@@ -0,0 +1,473 @@
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- What is client-side routing?
+- How do you use React Router to do client-side routing?
+- How do you create nested and dynamic paths?
+- How do you add a "catch-all" route?
+- How do you add protected routes?
+
+### Client-Side Routing
+
+Client-side routing is the type of routing where Javascript takes over the duty of handling the routes in an application. Client-side routing helps in building single-page applications (SPAs) without refreshing as the user navigates. For example, when a user clicks a navbar element, the URL changes and the view of the page is modified accordingly, within the client.
+
+Say you are cooking some chicken. If you want to cook it well and nice, you will have to:
+
+1. Put the chicken in the oven and set it to cook with appropriate time and heating
+2. Wait till the dish gives out that satisfying smell
+3. Start munching!
+
+This is common to all websites, you set the oven up for what you want (visit any URL, like [https://theodinproject.com/](https://theodinproject.com/)), wait for the oven to be done with the cooking (the loading screen), and tada, enjoy your delicious food (your page is ready for use). But what if you forgot to add some spices before you cooked it up? You have to repeat this flow again:
+
+1. Add the spices to the chicken
+2. Put it back into the oven and set it up to be reheated
+3. Wait for it to be nice and warm
+4. Now you can eat it!
+
+Here is where we reiterate, **the chicken needs to be reheated**. In a general multi-page application (MPAs), the browser reloads every time you click on a link to navigate. With client-side routing, **you never leave the page you are on** - you bring the microwave to the table to ensure that you don't run into the "missing spices" issues. The link requests are intercepted by the Javascript that you write, instead of letting them go directly to the server.
+
+### A Reactive Solution
+
+While client-side routing allows for nicer, app-like interactions (since you are controlling the routes, you can make fancy CSS animations across route changes), a lot of caveats can be missed. Browsers reloads notify screen-readers of new content to read, so you will need to notify screen-readers of route updates manually. However, with the help of a robust library, you can often address these concerns!
+
+React Router is a standard routing library for React applications. By using React Router, we can specify React components, that can be rendered based on the route, and so much more. Let's dive in!
+
+### Adding A Router
+
+Let's make a small app to understand how this router is implemented. Create a new project using `create-react-app` and get rid of the boilerplate code as you already learned in previous lessons (just leave the `index.js` and the `App.js` files). Let's add some mock pages as an example. Create a new `Profile.js` file with the following component:
```

Comment:
It's decided that we will be removing CRA from the new React course. So we should remove CRA from here. Suggested wording:
```suggestion
Let's make a small app to understand how this router is implemented. Create a new React project and let's add some mock pages as an example. Create a new `Profile.js` file with the following component:
```